src/python/grpcio/grpc/_cython/_cygrpc/call.pyx.pxi

```diff
@@ -106,6 +107,8 @@ cdef class Call:
   def __dealloc__(self):
     if self.c_call != NULL:
       grpc_call_destroy(self.c_call)
+    grpc_shutdown()
```

Comment:
Interesting...
I just looked at the code here 
https://github.com/grpc/grpc/blob/master/src/core/lib/surface/init.c

I'm guessing that if this approach "solves" memory leaks that were present because of PyPy garbage collection, it is only working because g_initializations == 0 never occurs after this change.  In which case, we could accomplish the same thing by just never calling grpc_shutdown.